feat: obfuscate data.secrets if present in experiment config [DET-10232] (#9635)
determined-ai/determined: docs/release-notes/secret-experiment-config.rst

:orphan:

**New Features**

> -  Experiments: In experiment config, subfields of ``data.secrets`` are obfuscated.

Reviewer: ```suggestion
- Experiments: Obfuscate subfields of ``data.secrets`` in the  :ref:`experiment configuration <experiment-config-data>`.
```

Reviewer: there was a request from Product to link back to docs (from release notes)
